File: backend/src/main/java/com/hr_solution/model/entities/Pergunta.java
package com.hr_solution.model.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.hr_solution.model.enuns.EixoPergunta;
import com.hr_solution.model.enuns.LiderancaPergunta;
import com.hr_solution.model.enuns.TipoPergunta;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "pergunta")
@Getter
@Setter
public class Pergunta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 1000)
    private String enunciado;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoPergunta tipo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = true)
    private EixoPergunta eixo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = true)    
    private LiderancaPergunta lideranca;

    //Peso da pergunta na avaliação
    @Column(nullable = false)  
    private Integer pesoTotal;

    @Column(nullable = false)
    private Integer NumeroAlternativas;

    @Column(nullable = false, length = 800)
    private String AlternativaUm;

    //Peso da alternativa na pergunta
    @Column(nullable = false)
    private Integer PesoUm;

    @Column(length = 800, nullable = false)
    private String AlternativaDois;

    @Column(nullable = false)
    private Integer PesoDois;

    @Column(length = 800,nullable = true)
    private String AlternativaTres;

    @Column(nullable = true)
    private Integer PesoTres;

    @Column(length = 800, nullable = true)
    private String AlternativaQuatro;
    
    @Column(nullable = true)
    private Integer PesoQuatro;

    @Column(length = 800, nullable = true)
    private String AlternativaCinco;

    @Column(nullable = true)
    private Integer PesoCinco;

    @Column(length = 800, nullable = true)
    private String AlternativaSeis;

    @Column(nullable = true)
    private Integer PesoSeis;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "avaliacao_id")
    @JsonIgnore
    private Avaliacao avaliacao;
}

File: frontend/src/model/entities/Pergunta.ts
import type { EixoPergunta } from "../enuns/EixoPergunta";
import type { LiderancaPergunta } from "../enuns/LiderancaPergunta";
import type { TipoPergunta } from "../enuns/TipoPergunta";
import type { Avaliacao } from "./Avaliacao";

export interface Pergunta {
  id?: number;
  enunciado: string;
  tipo: TipoPergunta;
  eixo?: EixoPergunta;
  lideranca?: LiderancaPergunta;
  ordem?: number;

  pesoTotal: number;
  numeroAlternativas: number;

  alternativaUm: string;
  pesoUm: number;

  alternativaDois: string;
  pesoDois: number;

  alternativaTres?: string;
  pesoTres?: number;

  alternativaQuatro?: string;
  pesoQuatro?: number;

  alternativaCinco?: string;
  pesoCinco?: number;

  alternativaSeis?: string;
  pesoSeis?: number;

  avaliacao?: Avaliacao;
}
